### PR TITLE
Support BESS protocol (for UML)

### PR DESCRIPTION
**Pull request by AkihiroSuda**
BESS protocol transferrs L2 packets as AF_UNIX SOCK_SEQPACKET .
BESS protocol has been used by the vector network interfaces of User Mode Linux (UML).

```
(terminal 1) $ slirp4netns --target-type=bess /tmp/bess.sock
(terminal 2) $ linux.uml vec0:transport=bess,dst=/tmp/bess.sock,depth=128,gro=1 root=/dev/root rootfstype=hostfs init=/bin/bash mem=2G
(terminal 2: UML)$ ip addr add 10.0.2.100/24 dev vec0
(terminal 2: UML)$ ip link set vec0 up
(terminal 2: UML)$ ip route add default via 10.0.2.2
```

More docs about the User Mode Linux with BESS socket transport: https://www.kernel.org/doc/html/latest/virt/uml/user_mode_linux_howto_v2.html#bess-socket-transport

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
cc @giuseppe @elmarco FYI

File: main.c [hunk 1]
```diff
@@ -667,9 +762,49 @@ static void parse_args(int argc, char *const argv[], struct options *options)
...
+        strcmp(options->target_type, "netns") != 0) {
+        fprintf(stderr, "--target-type must be either \"netns\" or \"bess\"\n");
+        goto error;
+    }
     if (argc - optind < 2) {
         goto error;
     }
+    if (argc - optind > 2) {
+        // not an error, for preventing potential compatibility issue
+        printf("WARNING: too many arguments\n");
```
Comment (giuseppe) [inline main.c:806]: should this go to stderr?

File: main.c [hunk 2]
```diff
@@ -229,6 +236,70 @@ static int child(int sock, pid_t target_pid, bool do_config_network,
...
+        perror("child_bess: bind");
+        goto error;
+    }
+    if (listen(bess_listenfd, 0) < 0) {
+        perror("child_bess: listen");
+        goto error;
+    }
+
+    /* Accept a connection, and send its connection to the parent */
+    printf("Waiting for connection to %s\n", bess_socket);
```
Comment (giuseppe) [inline main.c:281]: we are using both stdout and stderr for diagnostic messages.  Should we just stick to stderr?

**Review by giuseppe — APPROVED** [review]
I am not familiar with BESS but the code LGTM

Just left two minor comments

File: main.c (hunk 1 above)
Comment (AkihiroSuda) [reply main.c:806]: We have been using stdout or printing WARNING messages (why? 😅 )
https://github.com/rootless-containers/slirp4netns/blob/f2ae176d17db4ac044a8431e90cf8c9de5144584/main.c#L563-L604

File: main.c (hunk 2 above)
Comment (AkihiroSuda) [reply main.c]: fixed

**Comment by AkihiroSuda** [comment]
```
Step 9/19 : RUN  git pull && git checkout ${LIBSLIRP_COMMIT} && meson setup build && ninja -C build install
 ---> Running in f5c0822a3dc4
fatal: unable to access 'https://gitlab.freedesktop.org/slirp/libslirp.git/': The requested URL returned error: 504
The command '/bin/sh -c git pull && git checkout ${LIBSLIRP_COMMIT} && meson setup build && ninja -C build install' returned a non-zero code: 1
```

:cry:
https://github.com/rootless-containers/slirp4netns/runs/4723883332?check_suite_focus=true

**Pull request merged**